src/debugpy/common/util.py

```diff
@@ -148,3 +148,18 @@ def srcnameof(obj):
         name += ")"
 
     return name
+
+
+def skip_trace():
+    """Returns True if tracing should be skipped for the current thread."""
+    return "DEBUGPY_TRACE_DEBUGPY" not in os.environ
+
+
+def disable_tracing(thread):
+    """Disables tracing for the given thread if DEBUGPY_TRACE_DEBUGPY is not set.
+    DEBUGPY_TRACE_DEBUGPY is used to debug debugpy with debugpy
+    """
+    if skip_trace():
+        thread.pydev_do_not_trace = True
+        thread.is_pydev_daemon_thread = True
+        thread.daemon = True
```

Comment:
This one doesn't belong here - it affects the behavior of the thread when the process shuts down (whether it waits for this thread or not before exiting), not whether it's traced by the debugger.

Reply:
Okay, wasn't sure. I'll put it back.